js_modules/dagit/packages/core/src/assets/LaunchAssetChoosePartitionsDialog.tsx

```diff
@@ -370,9 +371,9 @@ const LaunchAssetChoosePartitionsDialogBody: React.FC<Props> = ({
   };
 
   const launchButton = () => {
-    if (launchAsBackfill && !canLaunchPartitionBackfill.enabled) {
+    if (launchAsBackfill && !canLaunchPartitionBackfill) {
       return (
-        <Tooltip content={canLaunchPartitionBackfill.disabledReason}>
+        <Tooltip content={disabledReasons.canLaunchPartitionBackfill}>
```

Comment:
This all looks great! I'm a big fan of removing this foot cannon... I wish Prettier would keep the destructuring to one line, but oh well.

One other thought -- I like the structure in this PR because we could potentially have other attributes on each permission in other dicts like `disabledReasons` BUT if it's just going to be these two, we could also flatten it all into a single field by flipping from enabled to disabled, and using the disabled data as the truthy values. 

(eg:  `disableLaunchPipelineExecution: {message: string} || false`)

I think I like the way you have it better though? At least, switching from "enabled = true" to "disabled = false" would be a more significant change!  😬

Reply:
Interesting thought! I think I'd be a little wary of representing the affirmative ability to take an action as a double-negative -- "I can launch this job, so it's `!disabled`" feels a little trickier to think through. (There are also other reasons why a control might be "disabled", so the negative version of this might be better represented as `cannotLaunchPipelineExecution`.)

I'll go ahead with this PR as is, and we can evaluate. :)